src/main/java/com/github/sh0ckr6/achievementborder/listeners/MobControl.java

```diff
@@ -0,0 +1,107 @@
+package com.github.sh0ckr6.achievementborder.listeners;
+
+import com.destroystokyo.paper.event.entity.EntityAddToWorldEvent;
+import com.destroystokyo.paper.event.entity.EntityRemoveFromWorldEvent;
+import com.github.sh0ckr6.achievementborder.AchievementBorder;
+import org.bukkit.Bukkit;
+import org.bukkit.WorldBorder;
+import org.bukkit.entity.Entity;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.Listener;
+
+import java.util.logging.Level;
+
+/**
+ * Listener event to control the number of hostile mobs
+ *
+ * @author gtaEPIC
+ * @since 1.1
+ */
+public class MobControl implements Listener {
+    /**
+     * Reference to the plugin
+     *
+     * @since 1.1
+     */
+    private final AchievementBorder plugin;
+
+    private int hostileMobs = 0; // Number of hostile mobs that are spawned
+
+    /**
+     * Registers this class as a {@link Listener} for the provided plugin
+     *
+     * @author gtaEPIC
+     * @since 1.1
+     */
+    public MobControl(AchievementBorder plugin) {
+        this.plugin = plugin;
+        plugin.getServer().getPluginManager().registerEvents(this, plugin);
+        for (Entity entity : Bukkit.getWorlds().get(0).getEntities()) {
+            onEntityAddToWorld(new EntityAddToWorldEvent(entity));
+        }
+    }
+
+    /**
+     * Calculates the number of Hostile Mobs should be allowed.
+     *
+     * @author gtaEPIC
+     * @since 1.1
+     * @return The max number of Hostile Mobs allowed
+     */
+    private int getMaxHostile() {
+        WorldBorder border = Bukkit.getWorlds().get(0).getWorldBorder();
+        double size = border.getSize();
+        if (size > 16) {
+            return (int) size;
+        }else return 0;
+    }
+
+    /**
+     * Checks to see if an entity is a hostile mob from the overworld
+     *
+     * @author gtaEPIC
+     * @since 1.1
+     * @param entity The entity to check
+     * @return True if hostile
+     */
+    private boolean isHostile(Entity entity) {
+        String[] hostile = {"Creeper", "Endermite", "Evoker", "Guardian", "Husk", "Silverfish", "Skeleton", "Slime",
+        "Stray", "Vex", "Vindicator", "Witch", "Zombie", "Zombie Villager"};
+        for (String name : hostile) {
+            if (entity.getName().equals(name)) return true;
+        }
+        return false;
```

Comment:
This entire check can be replaced with an `instanceof`. All hostile mobs are part of the `Monster` class, which extends `Entity`. You're actually also missing spiders with the manual check, which while considered passive during the day, are hostile in caves/at night.

```suggestion
        return entity instanceof Monster
```

Reply:
I was originally just going with mobs that were marked hostile (according to another site) hence why there were some mobs missing. If this is the way you would want it then javadocs will need to be fixed for this as well. (I'll do that in a sec)